### ISSUE TITLE
Connection reset by peer to API when cluster is unhealthy

### ISSUE THREAD

**Issue by splix** [etcd-io/etcd#17003]
### Bug report criteria

- [X] This bug report is not security related, security issues should be disclosed privately via security@etcd.io.
- [X] This is not a support request or question, support requests or questions should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
- [X] You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
- [ ] Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.

### What happened?

API doesn't respond if cluster is broken:

```
sudo curl -vv --cert /etc/ssl/etcd/ssl/member-master.pem --key /etc/ssl/etcd/ssl/member-master-key.pem --cacert /etc/ssl/etcd/ssl/ca.pem  https://localhost:2379/v2/members
*   Trying 127.0.0.1:2379...
* Connected to localhost (127.0.0.1) port 2379 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/etcd/ssl/ca.pem
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
....
* OpenSSL SSL_connect: Connection reset by peer in connection to localhost:2379
```

Which forks fine when cluster is healthy.

### What did you expect to happen?

API works and returns a [probably invalid] list of members.

### How can we reproduce it (as minimally and precisely as possible)?

1. Run a cluster with 3 nodes
2. Change IP on all those servers and restart the cluster (with updated config using only new IPs)
3. It looses quorum and cannot recover (cannot connect to old IPs, not accepting new IPs)
4. API doesn't work

### Etcd version (please run commands below)

<details>

```console
$ etcd --version
etcd Version: 3.5.9
Git SHA: bdbbde998
Go Version: go1.19.9
Go OS/Arch: linux/amd64

$ etcdctl version
etcdctl version: 3.5.9
API version: 3.5
```

</details>

### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>

### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>

**Comment by tjungblu** [comment]
> Change IP on all those servers and restart the cluster

I don't think that's a scenario that works with etcd. 

Did you try updating the member IPs? There's good docs here:
https://etcd.io/docs/v3.5/op-guide/runtime-configuration/#update-a-member

Alternatively, you need to remove the member with the old IP, restart with new IP, add the new one as a member. Rinse and repeat.

**Comment by tjungblu** [comment]
I don't think you can give a proper member listing response without quorum. Do you want to see a better error message for that case or what do you expect to happen?

**Comment by splix** [comment]
Maybe it can give a proper JSON with an empty array, or last known list? But that's not the point. Sitting on the connection without giving _any_ response doesn't look right. 

So yes, as you said, a better error message. An HTTP response with a non-200 status is fine. Right now it just blocks the connection, so from the API caller it's hard to guess what is really happening, because it can be blocked by a firewall, or something else.

And I actually mean any request, not just the "members list"; as I remember `/health` behaves the same way.

**Comment by moficodes** [comment]
I can take a look. Please assign this to me.
